File: sample-java/src/main/java/com/android/mvvmarchitecturejavasample/data/WanAndroidApiService.java
package com.android.mvvmarchitecturejavasample.data;

import com.android.mvvmarchitecturejavasample.entity.BaseEntity;
import com.android.mvvmarchitecturejavasample.entity.DemoEntity;

import java.util.List;

import io.reactivex.Observable;
import retrofit2.Call;
import retrofit2.http.GET;

public interface WanAndroidApiService {
    @GET("wxarticle/chapters/json")
    Observable<BaseEntity<List<DemoEntity>>> demoGet();
    @GET("https://www.google.com")
    Call<BaseEntity<List<DemoEntity>>> demoGet2();
}

File: sample/src/main/java/com/imyyq/sample/data/source/http/service/WanAndroidApiService.kt
package com.imyyq.sample.data.source.http.service

import com.imyyq.sample.entity.BaseEntity
import com.imyyq.sample.entity.DemoEntity
import com.imyyq.sample.entity.FriendWebSiteEntity
import io.reactivex.Observable
import okhttp3.ResponseBody
import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Query

interface WanAndroidApiService {
    // wanandroid 的开发 api 模拟登录 https://www.wanandroid.com/
    @GET("friend/json")
    suspend fun login(
        @Query("userName") userName: String,
        @Query("pwd") pwd: String
    ): BaseEntity<List<FriendWebSiteEntity?>?>?

    @GET("https://github.com/")
    fun google(
    ): Call<ResponseBody?>?

    @GET("wxarticle/chapters/json")
    fun demoGet(): Observable<BaseEntity<List<DemoEntity?>?>?>

}